packages/sdk/src/browser-wallets/leather/__tests__/index.spec.ts

```diff
@@ -0,0 +1,388 @@
+// @vitest-environment happy-dom
+
+import { Psbt } from "bitcoinjs-lib";
+
+import {
+  BrowserWalletExtractTxFromNonFinalizedPsbtError,
+  BrowserWalletNetworkMismatchError,
+  BrowserWalletRequestCancelledByUserError,
+} from "../../..";
+import { getAddresses, isInstalled, signMessage, signPsbt } from "..";
+import { LeatherErrorResponse } from "../utils";
+
+class LeatherError extends Error {
+  id: string;
+
+  jsonrpc: string;
+
+  error: {
+    code: number;
+    message: string;
+  };
+
+  constructor(err: LeatherErrorResponse) {
+    super(err.error.message);
+    this.id = err.id;
+    this.jsonrpc = err.jsonrpc;
+    this.error = err.error;
+  }
+}
```

Comment:
Are leather errors `instanceof Error`? It was not the case for Unisat. We should not create a class if it's not the same.

Instead of throwing LeatherError, we can assert the object being thrown

Reply:
oh you're right, updated